### PR TITLE
ligo: 0.47.0 -> 0.53.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

The release of 0.53.0 is now done so it would be good if we could land this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/ligo/default.nix [hunk 1]
```diff
@@ -2,41 +2,46 @@
...
+ocamlPackages.buildDunePackage rec {
   pname = "ligo";
-  version = "0.47.0";
+  version = "0.53.0-dev";
   src = fetchFromGitLab {
     owner = "ligolang";
     repo = "ligo";
-    rev = version;
-    sha256 = "sha256-VJs0pKA99mZXhipK1bSAZmgAHvYbGbjxdI2XyJYjSm0=";
+    rev = "2c2a029646050077e771be5485ba5a176bc8657f";
```
Comment (ulrikstrid) [inline pkgs/development/compilers/ligo/default.nix:16]: Will change this and the version as soon as it's tagged

File: pkgs/development/compilers/ligo/default.nix [hunk 2]
```diff
@@ -80,23 +90,23 @@ coq.ocamlPackages.buildDunePackage rec {
...
 
   checkInputs = [
     cacert
-    coq.ocamlPackages.ca-certs
+    ocamlPackages.ca-certs
   ];
 
   doCheck = false; # Tests fail, but could not determine the reason
 
-  patches = [ ./ligo.patch ]; # fix for core >= 0.15.0
```
Comment (ulrikstrid) [inline pkgs/development/compilers/ligo/default.nix:92]: Hoping that we land https://gitlab.com/ligolang/ligo/-/merge_requests/2061 before the release, I'm working with the devs to make it happen.

**Comment by ulrikstrid** [comment]
This is not ready for merge since LIGO has not tagged the release yet, but if I could get a review before then we should easily be able to just set the correct version etc and push it through when it's released.

File: pkgs/development/compilers/ligo/default.nix (hunk 2 above)
Comment (ulrikstrid) [reply pkgs/development/compilers/ligo/default.nix:92]: This has landed

File: pkgs/development/compilers/ligo/default.nix (hunk 1 above)
Comment (ulrikstrid) [reply pkgs/development/compilers/ligo/default.nix]: Done

**Comment by ulrikstrid** [comment]
@vbgl this is ready for review/merge if you have time to review the OCaml changes

**Comment by ulrikstrid** [comment]
With all the other PRs going in this only have 1 commit now, could you please review @vbgl ? :smile:

**Comment by ulrikstrid** [comment]
Rebased on top of #197039

**Review by Gau-thier — APPROVED** [review]

**Comment by ulrikstrid** [comment]
Rebased on `master` again, should be good to go

File: pkgs/development/compilers/ligo/default.nix [hunk 3]
```diff
@@ -2,41 +2,46 @@
 , fetchFromGitLab
 , git
 , coq
+, ocamlPackages
 , cacert
 , ocaml-crunch
 }:
 
-coq.ocamlPackages.buildDunePackage rec {
+ocamlPackages.buildDunePackage rec {
```
[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/compilers/ligo/default.nix:10]: Why was this constraint there in the first place? (i.e., why do we care that Coq and ligo are built with the same OCaml?)

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -8338,6 +8338,7 @@ with pkgs;
 
   ligo = callPackage ../development/compilers/ligo {
     coq = coq_8_14;
+    ocamlPackages = ocaml-ng.ocamlPackages_4_14;
```
Comment (vbgl) [inline pkgs/top-level/all-packages.nix:8341]: This change seem superfluous.

File: pkgs/development/compilers/ligo/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply pkgs/development/compilers/ligo/default.nix:10]: https://github.com/NixOS/nixpkgs/pull/141471#discussion_r729309120
You suggested it to me when I first added LIGO

**Pull request merged**